pkg/plugins/manager/loader/angulardetector/inspector.go

```diff
@@ -0,0 +1,45 @@
+package angulardetector
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/grafana/grafana/pkg/plugins"
+)
+
+// Inspector can inspect a plugin and determine if it's an Angular plugin or not.
+type Inspector interface {
+	// Inspect takes a plugin and checks if the plugin is using Angular.
+	Inspect(ctx context.Context, p *plugins.Plugin) (bool, error)
+}
+
+// PatternsListInspector is an Inspector that matches a plugin's module.js against all the patterns returned by
+// the detectorsProvider, in sequence.
+type PatternsListInspector struct {
+	// detectorsProvider returns the detectors that will be used by Inspect.
+	detectorsProvider detectorsProvider
+}
+
+func (i *PatternsListInspector) Inspect(ctx context.Context, p *plugins.Plugin) (isAngular bool, err error) {
+	f, err := p.FS.Open("module.js")
+	if err != nil {
```

Comment:
Nit not sure who calls this and what it does with the error. Maybe return false without an error if module.js doesn't exist - there are some backend plugins that might not have a module.js?